### PR TITLE
feat: Support the Count aggregation query

### PR DESCRIPTION
**Pull request by Mariatta**
Thank you for opening a Pull Request! Before submitting your PR, there are a few things you can do to make sure it goes smoothly:
- [x] Make sure to open an issue as a [bug/issue](https://github.com/googleapis/python-firestore/issues/new/choose) before writing your code!  That way we can discuss the change, evaluate designs, and agree on the general idea
- [x] Ensure the tests and linter pass
- [x] Code coverage does not decrease (if any source code was changed)
- [x] Appropriate docs were updated (if necessary)

Fixes #<issue_number_goes_here> 🦕

### CODE REVIEW

**Review by danieljbruce — COMMENTED** [review]
First pass.

File: tests/system/test_system.py [hunk 1]
```diff
@@ -536,6 +536,13 @@ def query_docs(client):
         operation()
 
 
+@pytest.fixture
```
Comment (kolea2) [inline tests/system/test_system.py:539]: looks good. can we add some tests running this with a transaction?

**Review by kolea2 — DISMISSED** [review]

File: tests/system/test_system.py [hunk 2]
```diff
@@ -1617,3 +1624,130 @@ def test_repro_391(client, cleanup):
...
+
+
+def test_count_query_get_multiple_aggregations(query):
+    count_query = query.count(alias="total").count(alias="all")
+
+    result = count_query.get()
+    assert len(result[0]) == 2
+
+    for r in result[0]:
+        assert r.alias in ["total", "all"]
```
[the remark this answers is not in the document]
Comment (Mariatta) [reply tests/system/test_system.py]: From my tests, the order of the result has been inconsistent. It doesn't always return it in the order of "total" and then "all".

I've now adjusted this test to assert that:
- there are two elements returned in the result
- the returned result has two values and they are both unique
- the result contains both aliases: "total" and "all"

File: tests/system/test_system.py (hunk 1 above)
Comment (Mariatta) [reply tests/system/test_system.py:539]: I've added a new test that uses transaction in both test_system and test_system_async.

File: google/cloud/firestore_v1/async_aggregation.py [hunk 3]
```diff
@@ -0,0 +1,122 @@
...
+        return result
+
+    async def stream(
+        self,
+        transaction=None,
+        retry: Union[
+            retries.Retry, None, gapic_v1.method._MethodDefault
+        ] = gapic_v1.method.DEFAULT,
+        timeout: float | None = None,
+    ):  # -> Generator[document.DocumentSnapshot, Any, None]: AsyncGenerator
```
Comment (kolea2) [inline google/cloud/firestore_v1/async_aggregation.py:84]: should this be commented out?

File: google/cloud/firestore_v1/base_aggregation.py [hunk 4]
```diff
@@ -0,0 +1,199 @@
+from __future__ import annotations
```
Comment (kolea2) [inline google/cloud/firestore_v1/base_aggregation.py:1]: missing header

**Review by kolea2 — COMMENTED** [review]
lgtm to me after header fix and PTAL at last comment.

Would like someone from yoshi-python to review as I've only looked from an API perspective. Thanks!

File: google/cloud/firestore_v1/base_aggregation.py (hunk 4 above)
Comment (Mariatta) [reply google/cloud/firestore_v1/base_aggregation.py:23]: Added.

File: google/cloud/firestore_v1/async_aggregation.py (hunk 3 above)
Comment (Mariatta) [reply google/cloud/firestore_v1/async_aggregation.py]: It actually should be uncommented so we have type annotations. I have fixed this.

File: google/cloud/firestore_v1/aggregation.py [hunk 5]
```diff
@@ -0,0 +1,158 @@
...
+
+    def _get_stream_iterator(self, transaction, retry, timeout):
+        """Helper method for :meth:`stream`."""
+        request, kwargs = self._prep_stream(
+            transaction,
+            retry,
+            timeout,
+        )
+
+        response_iterator = self._client._firestore_api.run_aggregation_query(
```
Comment (nicain) [inline google/cloud/firestore_v1/aggregation.py:85]: Nit: return directly here, no need for the temporary variable `response_iterator`

Comment (Mariatta) [reply google/cloud/firestore_v1/aggregation.py]: Thanks, updated.

File: tests/system/test_system.py [hunk 6]
```diff
@@ -46,11 +47,13 @@ def _get_credentials_and_project():
     if FIRESTORE_EMULATOR:
         credentials = EMULATOR_CREDS
         project = FIRESTORE_PROJECT
-    else:
+    elif FIRESTORE_CREDS:
         credentials = service_account.Credentials.from_service_account_file(
             FIRESTORE_CREDS
         )
         project = FIRESTORE_PROJECT or credentials.project_id
+    else:
```
Comment (kolea2) [inline tests/system/test_system.py:55]: would like someone from @googleapis/yoshi-python to take a look at this specifically

**Review by kolea2 — APPROVED** [review]
lgtm after @googleapis/yoshi-python takes a look as well. Thanks!

File: tests/system/test_system.py (hunk 6 above)
Comment (meredithslota) [reply tests/system/test_system.py:55]: @kolea2 did anyone from @googleapis/yoshi-python review this? I can't tell from the conversation.

Comment (kolea2) [reply tests/system/test_system.py:55]: I believe @nicain did!

**Pull request merged**